File: netlogo-headless/src/main/prim/etc/_linkshapes.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.LogoListBuilder
import org.nlogo.core.LogoList
import org.nlogo.nvm.{ Context, Reporter }

class _linkshapes extends Reporter {
  override def report(context: Context): LogoList = {
    val result = new LogoListBuilder
    for(shape <- world.linkShapeList.shapes)
      result.add(shape.name)
    result.toLogoList
  }
}

File: netlogo-gui/src/main/prim/etc/_linkshapes.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc;

import java.util.List;

import org.nlogo.core.LogoList$;
import org.nlogo.core.Shape;
import org.nlogo.api.LogoListBuilder;
import org.nlogo.core.Syntax;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;

import scala.collection.Iterator;

public final class _linkshapes
    extends Reporter {


  @Override
  public Object report(Context context) {
    Iterator<Shape> shapes = world.linkShapeList().shapes().iterator();
    LogoListBuilder b = new LogoListBuilder();
    while (shapes.hasNext()) {
      b.add(shapes.next().name());
    }
    return b.toLogoList();
  }
}
